Tools/LyTestTools/ly_test_tools/environment/file_system.py

```diff
@@ -182,7 +182,26 @@ def untgz(dest, src, exact_tgz_size=False, force=False, allow_exists=False):
 
         # Extract it and return final path.
         start_time = time.time()
-        tar_file.extractall(dst_path)
+        def is_within_directory(directory, target):
+            
+            abs_directory = os.path.abspath(directory)
+            abs_target = os.path.abspath(target)
+        
+            prefix = os.path.commonprefix([abs_directory, abs_target])
+            
+            return prefix == abs_directory
+        
+        def safe_extract(tar, path=".", members=None, *, numeric_owner=False):
+        
+            for member in tar.getmembers():
+                member_path = os.path.join(path, member.name)
+                if not is_within_directory(path, member_path):
+                    raise Exception("Attempted Path Traversal in Tar File")
```

Comment:
Not generally recommended to use 'Exception' as it is too general. In this case, this looks more like a ValueError